rust-sdk/src/poller.rs

```diff
@@ -80,14 +90,16 @@ impl PollerThread {
     /// # }
     /// ```
     pub fn wait_for_configuration(&self) -> Result<()> {
-        self.0.wait_for_configuration()
+        self.thread
+            .runtime()
+            .block_on(self.poller.wait_for_configuration())
```

Comment:
Another nicety of the background runtime is that it allows us running async functions from sync context (see [Handle::block_on](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.block_on)). This is generally discouraged because it blocks user's thread but sometimes we actually want this blocking (like here).